client.go

```diff
@@ -22,47 +23,30 @@ import (
  * Client structure as interface implementation
  */
 type Client struct {
-	cancel       func()
-	ctx          context.Context
-	ch           chan []*models.EventModel
-	chUser       chan []*models.UserModel
-	chCompany    chan []*models.CompanyModel
-	flush        chan chan struct{}
-	interval     time.Duration
-	timeout      time.Duration
-	responseETag string
-	rulesETag    string
-}
-
-func (c *Client) GetETag() string {
-	return c.responseETag
-}
-
-/*
- * Get X-Moesif-Config-Etag from response headers
- */
-func (c *Client) fetchETagFromHeader(headers http.Header) {
-	if eTag, ok := headers["X-Moesif-Config-Etag"]; ok {
-		c.responseETag = eTag[0]
-	}
-	if eTag, ok := headers["X-Moesif-Rules-Tag"]; ok {
-		c.rulesETag = eTag[0]
-	}
+	cancel               func()
+	ctx                  context.Context
+	ch                   chan []*models.EventModel
+	chUser               chan []*models.UserModel
+	chCompany            chan []*models.CompanyModel
+	flush                chan chan struct{}
+	interval             time.Duration
+	timeout              time.Duration
```

Comment:
we no longer need responseETag or rulesETag?

Reply:
Yep! previously concurrent goroutines just raced to overwrite those variables and eventually hopefully converged on the correct latest value, but now callbacks handle each response e-tag individually so no need to unsynchronized global state